src/main/java/com/rocketden/main/model/problem/Problem.java

```diff
@@ -52,4 +60,9 @@ public void addTestCase(ProblemTestCase testCase) {
     public boolean removeTestCase(ProblemTestCase testCase) {
         return testCases.remove(testCase);
     }
+
+    public void addProblemInput(ProblemInput problemInput) {
+        problemInputs.add(problemInput);
+        problemInput.setProblem(this);
+    }
```

Comment:
nit: This is pretty small but could you add a quick test for this in ProblemEntityTests?

Reply:
Sure, no problem -- done!